trace/rate_limit.go

```diff
@@ -146,10 +145,19 @@ func (r *SharedRateLimit) startNewEpoch(epochStart time.Time) {
 		r.intervalTimer.Reset(randomOffset)
 	}
 
+	r.lock.Unlock()
```

Comment:
Seems okay, though to make this more robust against future changes, I'd feel more comfortable using `defer`, perhaps in a nested anonymous function.